anx/provider/loadbalancer/reconciliation/reconcile_binds.go

```diff
@@ -12,27 +12,27 @@ import (
 
 const bindResourceTypeIdentifier = "bd24def982aa478fb3352cb5f49aab47"
 
-func (r *reconciliation) storePublicAddress(addr string) {
-	if idx := sort.SearchStrings(r.publicAddresses, addr); idx >= len(r.publicAddresses) || r.publicAddresses[idx] != addr {
-		r.publicAddresses = append(r.publicAddresses, addr)
-		sort.Strings(r.publicAddresses)
+func (r *stateRetrieverImpl) storePublicAddress(lbID, addr string) {
+	if idx := sort.SearchStrings(r.loadBalancers[lbID].state.publicAddresses, addr); idx >= len(r.loadBalancers[lbID].state.publicAddresses) || r.loadBalancers[lbID].state.publicAddresses[idx] != addr {
+		r.loadBalancers[lbID].state.publicAddresses = append(r.loadBalancers[lbID].state.publicAddresses, addr)
+		sort.Strings(r.loadBalancers[lbID].state.publicAddresses)
 	}
 }
 
-func (r *reconciliation) filterBinds(allBinds []*lbaasv1.Bind) ([]*lbaasv1.Bind, error) {
+func (r *stateRetrieverImpl) filterBinds(lbID string, allBinds []*lbaasv1.Bind) ([]*lbaasv1.Bind, error) {
```

Comment:
another candidate to move into `state_retriever.go` - after all, this filters the state to only contain `Binds` for our LoadBalancers.